src/algebra/fields/bigint.hpp

```diff
@@ -50,6 +50,17 @@ class bigint {
     void to_mpz(mpz_t r) const;
     bool test_bit(const std::size_t bitno) const;
 
+    template<mp_size_t m> inline bigint<n> operator+(const bigint<m>& other) const;
```

Comment:
These aren't template-templates, but the ones in the .tcc are, should they be consistent? Why are they template-templates?